### PR TITLE
Feature/partial labels

### PR DESCRIPTION
**Pull request by Z30G0D**
partial labels

### CODE REVIEW

File: src/conf/data/defaults.yaml [hunk 1]
```diff
@@ -6,22 +6,23 @@ data:
   data_sample_rate: 44100
```
Comment (noamzbr) [inline src/conf/data/defaults.yaml:6]: Don't think we want to commit changes to this file

File: src/conf/data/partial_label.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    preprocessors: ${_preprocessors}
+    seq_length: 0.5
+    len_buffer: 0.1
+    margin_ratio: 0.5
+    data_sample_rate: ${data.data_sample_rate}
+    sample_rate: ${data.sample_rate}
+    equalize_data: true
+    slice_flag: true
+    partial_positive: true
+  val_dataset:
```
Comment (noamzbr) [inline src/conf/data/partial_label.yaml:23]: why for val partials aren't considered positive?

File: src/conf/data/partial_label.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    _target_: data.ClassifierDataset
+    data_path: '../../finding_willy/recordings_2018_filtered/'
+    mode: train 
+    metadata_path: '../annotations/combined_annotations_filtered_train.csv'
+    augmentations_p: 0.8
+    augmentations: ${_augmentations}
+    preprocessors: ${_preprocessors}
+    seq_length: 0.5
+    len_buffer: 0.1
+    margin_ratio: 0.5
```
Comment (noamzbr) [inline src/conf/data/partial_label.yaml:17]: Sorry for silly question - but doesn't this behavior kinda overlap with partial positives? Can you remind us what the margin_ratio does?

File: src/data.py [hunk 4]
```diff
@@ -119,14 +122,49 @@ def _slice_sequence(self, len_buffer):
...
         duplicate_size_vector = [len(list_elem) for list_elem in new_begin_time] # vector to duplicate original dataframe
         new_begin_time = np.concatenate(new_begin_time) # vectorize to array
         new_end_time = np.concatenate(list(x[1:] for x in sliced_times)) # same for end_times
         self.metadata = self.metadata.iloc[self.metadata.index.repeat(duplicate_size_vector)].reset_index(drop=True)
         self.metadata['begin_time'] = new_begin_time
         self.metadata['end_time'] = new_end_time
-        self.metadata['call_length'] = np.shape(self.metadata)[0] * [self.seq_length]
+        new_call_length = self.metadata['end_time'] - self.metadata['begin_time']
+        self.metadata['call_length'] = new_call_length
+        self.metadata = self.metadata[self.metadata['call_length'] > self.seq_length]
```
Comment (noamzbr) [inline src/data.py:167]: don't we want >= ?

**Comment by Z30G0D** [comment]
Following last week's conversation and silly changes...
As Noam mentioned,  I just changed margin_ratio a bit to include the label condition.

File: src/data.py [hunk 5]
```diff
@@ -142,8 +147,8 @@ def _get_audio(self, path_to_file, begin_time, end_time):
         output:
         audio - pytorch tensor (1-D array)
         """
-        if self.mode == "train":
-            if self.margin_ratio != 0:
+        if (self.mode == "train") and (label == 1):
```
Comment (mosheman5) [inline src/data.py:150]: important bugfix!

File: src/data.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
 from torchvision import transforms
 from data_augmentation import ChainedAugmentations
 import torchaudio
+from copy import deepcopy
```
Comment (mosheman5) [inline src/data.py:11]: double import?

File: src/data.py [hunk 7]
```diff
@@ -38,12 +39,15 @@ def __init__(self, data_path, metadata_path, augmentations, augmentations_p, pre
         self.metadata = pd.read_csv(self.metadata_path)
         self.mode = mode
         self.seq_length = seq_length
+        # self.partial_positive = partial_positive
```
Comment (mosheman5) [inline src/data.py:42]: please delete if not needed

File: src/data.py [hunk 8]
```diff
@@ -38,12 +39,15 @@ def __init__(self, data_path, metadata_path, augmentations, augmentations_p, pre
...
+        # self.partial_positive_coeff = 0.5
         self.sample_rate = sample_rate
         self.data_sample_rate = data_sample_rate
         self.sampler = torchaudio.transforms.Resample(orig_freq=data_sample_rate, new_freq=sample_rate)
         self._preprocess_metadata(len_buffer, equalize_data, slice_flag)
         self.augmenter = self._set_augmentations(augmentations, augmentations_p)
         self.preprocessor = self.set_preprocessor(preprocessors)
+        assert (0 <= margin_ratio) and (1 >= margin_ratio)
```
Comment (mosheman5) [inline src/data.py:50]: me like assertions :)

File: src/conf/data/defaults.yaml [hunk 9]
```diff
@@ -6,19 +6,19 @@ data:
...
     preprocessors: ${_preprocessors}
-    seq_length: 1 
+    seq_length: 1
     len_buffer: 0.1
     margin_ratio: 0.5
     data_sample_rate: ${data.data_sample_rate}
     sample_rate: ${data.sample_rate}
     equalize_data: true
-    slice_flag: false
+    slice_flag: true
```
Comment (mosheman5) [inline src/conf/data/defaults.yaml:21]: why?

**Review by mosheman5 — CHANGES_REQUESTED** [review]
Improtant bugfix!
had tiny comments and potential bug in the data conf
other than that, lgtm
